apollo-router/src/error.rs

```diff
@@ -582,6 +562,37 @@ impl IntoGraphQLErrors for ValidationErrors {
                             })
                             .unwrap_or_default(),
                     )
+                    .extension_code("GRAPHQL_PARSING_FAILED")
+                    .build()
+            })
+            .collect())
+    }
+}
+
+/// Collection of schema validation errors.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub(crate) struct ValidationErrors {
+    pub(crate) errors: Vec<apollo_compiler::execution::GraphQLError>,
```

Comment:
Should this use `GraphQLError` or `DiagnosticsList`? 🤔  cc @SimonSapin

Reply:
we need something that implements `Serialize` and `Deserialize` so I used `GraphQLError`

Reply:
For that purpose either works. `DiagnosticList` can be converted to `Vec<GraphQLError>` either when constructing this struct or later when using it. Doing it later allows printing with nice CLI formatting in `dbg!`, but requires carrying a `SourceMap` around somehow.